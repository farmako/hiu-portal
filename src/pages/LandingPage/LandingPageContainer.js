import { connect } from 'react-redux';
import LandingPage from './LandingPage';
import { onCreateConsentResetState } from '../../redux/actions/createConsentActions';
import { onSearchResetState } from '../../redux/actions/searchPatientIdActions';
import { loadConsents } from '../../redux/actions/loadConsentsActions';
import { fetchValueSetsRequest as loadConfigValueSets } from '../../redux/actions/initAppActions';
import { fetchConfigRequest as loadCMConfigurations } from '../../redux/actions/configAppActions';
import getNestedObject from '../../utils/getNestedObject';

const mapStateToProps = (state) => ({
  success: getNestedObject(state, 'createConsent.success'),
});

const mapDispatchToProps = {
  onCreateConsentResetState,
  onSearchResetState,
  loadConsents,
  loadConfigValueSets,
  loadCMConfigurations,
};

export default connect(mapStateToProps, mapDispatchToProps)(LandingPage);
