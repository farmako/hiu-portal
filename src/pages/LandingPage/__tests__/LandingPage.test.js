import React from 'react';
import { shallow } from 'enzyme';
import Button from '@material-ui/core/Button';
import { Modal, Snackbar } from '@material-ui/core';
import IconButton from '@material-ui/core/IconButton';
import LandingPage from '../LandingPage';

describe('Landing Page', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<LandingPage />);
  });

  it('should render properly', () => {
    expect(wrapper.debug()).toMatchSnapshot();
  });

  it('Request Access: Modal, should be closed by default', () => {
    expect(wrapper.find(Modal).props().open).toBe(false);
  });

  it('Success Message: Snackbar, should be closed by default', () => {
    expect(wrapper.find(Snackbar).props().open).toBe(false);
  });

  it('New Consent Request: Button, should open Request Access modal on click', () => {
    wrapper.find(Button).props().onClick();
    expect(wrapper.find(Modal).props().open).toBe(true);
  });

  describe('After successful creation of consent', () => {
    let onCreateConsentResetState;
    let onSearchResetState;
    let loadConsents;
    let loadConfigValueSets;
    let landingPageWrapper;

    beforeAll(() => {
      onCreateConsentResetState = jest.fn();
      onSearchResetState = jest.fn();
      loadConsents = jest.fn();
      loadConfigValueSets = jest.fn();
      landingPageWrapper = shallow(
        <LandingPage
          success
          onCreateConsentResetState={onCreateConsentResetState}
          onSearchResetState={onSearchResetState}
          loadConsents={loadConsents}
          loadConfigValueSets={loadConfigValueSets}
        />
      );
      landingPageWrapper.find(Button).props().onClick(); // Open Modal
    });

    it('Success Message: Snackbar, should popup after successfully requesting a consent', () => {
      const successMessageSnackbar = landingPageWrapper.find(Snackbar);
      expect(successMessageSnackbar.props().open).toBe(true);
      expect(successMessageSnackbar.props().message).toBe(
        'Consent request initiated!'
      );
    });

    it('Success Message: Snackbar, should close popup on click of close icon', () => {
      landingPageWrapper.find(Snackbar).props().onClose();
      expect(landingPageWrapper.find(Snackbar).props().open).toBe(false);
    });

    it('should reload the consent list', () => {
      expect(loadConsents).toHaveBeenCalled();
    });

    it('should reset previously created consent creation state on close', () => {
      landingPageWrapper.find(Modal).props().onClose();
      expect(onCreateConsentResetState).toHaveBeenCalled();
    });

    it('should reset the previously searched patient state on close', () => {
      landingPageWrapper.find(Modal).props().onClose();
      expect(onSearchResetState).toHaveBeenCalled();
    });

    it('should reset the modal state on close button click', () => {
      landingPageWrapper.find(IconButton).simulate('click');
      expect(onSearchResetState).toHaveBeenCalled();
      expect(onCreateConsentResetState).toHaveBeenCalled();
    });
  });
});
