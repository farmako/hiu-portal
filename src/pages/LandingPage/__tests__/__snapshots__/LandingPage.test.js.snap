// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Landing Page should render properly 1`] = `
"<div>
  <WithStyles(ForwardRef(Snackbar)) anchorOrigin={{...}} open={false} autoHideDuration={4000} onClose={[Function: handleSnackBarClose]} message=\\"Consent request initiated!\\" />
  <Header />
  <WithStyles(ForwardRef(Button)) className=\\"makeStyles-button-2\\" type=\\"button\\" variant=\\"contained\\" onClick={[Function: handleOpen]} color=\\"primary\\">
    New Consent Request
  </WithStyles(ForwardRef(Button))>
  <ForwardRef(Modal) open={false} onClose={[Function: handleClose]}>
    <div style={{...}} className=\\"makeStyles-paper-1\\">
      <WithStyles(ForwardRef(IconButton)) aria-label=\\"close\\" onClick={[Function: handleClose]} className=\\"makeStyles-closeButton-3\\">
        <CancelIcon />
      </WithStyles(ForwardRef(IconButton))>
      <Connect(RequestAccess) />
    </div>
  </ForwardRef(Modal)>
  <Connect(ConsentsListTable) />
</div>"
`;
